SQLAssignment.sql.sql

```diff
@@ -0,0 +1,1043 @@
++------+----------+--------+
+| CNUM | CNAME    | RATING |
++------+----------+--------+
+| 2001 | Hoffman  |    100 |
+| 2002 | Giovanni |    200 |
+| 2003 | Liu      |    200 |
+| 2004 | Grass    |    300 |
+| 2006 | Clemens  |    100 |
+| 2008 | Cisneros |    300 |
+| 2007 | Pereira  |    100 |
++------+----------+--------+
+36) Write a query that totals the orders for each day and places the results in descending order.
+mysql> SELECT ODATE,SUM(AMT) AS Total_Amount FROM Orders GROUP BY ODATE ORDER BY Total_Amount;
++------------+--------------+
+| ODATE      | Total_Amount |
++------------+--------------+
+| 1990-10-04 |      1788.98 |
+| 1990-10-05 |      4723.00 |
+| 1990-10-03 |      8944.59 |
+| 1990-10-06 |     11201.83 |
++------------+--------------+
+37) Write a select command that produces the rating followed by the name of each customer in SanJose.
+mysql>  SELECT c.CNAME,c.RATING FROM Salespeople s,customers c WHERE s.SNUM = c.SNUM AND s.CITY = 'SanJose';
++-------+--------+
+| CNAME | RATING |
++-------+--------+
+| Liu   |    200 |
+| Grass |    300 |
++-------+--------+
+38) Find all orders with amounts smaller than any amount for a customer in SanJose.
+mysql> SELECT ONUM,AMT,ODATE,CNUM,SNUM 
+       FROM Orders WHERE AMT <(SELECT MAX(AMT) 
+                               FROM Orders WHERE SNUM = (SELECT SNUM 
+														FROM Salespeople WHERE CITY ='SanJose'));
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-10-03 | 2008 | 1007 |
+| 3003 |  767.19 | 1990-10-03 | 2001 | 1001 |
+| 3002 | 1900.10 | 1990-10-03 | 2007 | 1004 |
+| 3006 | 1098.16 | 1990-10-03 | 2008 | 1007 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3007 |   75.75 | 1990-10-04 | 2004 | 1002 |
+| 3008 | 4723.00 | 1990-10-05 | 2006 | 1001 |
+| 3010 | 1309.95 | 1990-10-06 | 2004 | 1002 |
++------+---------+------------+------+------+
+39) Find all orders with above average amounts for their customers.
+mysql> select onum,AMT,ODATE,CNUM,SNUM from orders where amt >(select avg(amt) from orders);
++------+---------+------------+------+------+
+| onum | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3008 | 4723.00 | 1990-10-05 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
++------+---------+------------+------+------+
+40) Write a query that selects the highest rating in each city.
+mysql> SELECT CITY, MAX(RATING) FROM Customers GROUP BY CITY;
++---------+-------------+
+| CITY    | MAX(RATING) |
++---------+-------------+
+| London  |         100 |
+| Rome    |         200 |
+| SanJose |         300 |
+| Berlin  |         300 |
++---------+-------------+
+41) Write a query that calculates the amount of the salesperson’s commission on each order by a customer with a rating above 100.00.
+
+mysql> SELECT SNAME,COMM*AMT AS Amount 
```

Comment:
This is a wrong query. The commission amount should be calculated using the below formula:
Commission Amount = (Amt*Com)/100. Please update your query, here we are finding commission and it can't be such a big number :-)